src/rust/lib/system/web/src/event/listener.rs

```diff
@@ -0,0 +1,100 @@
+use crate::prelude::*;
+
+use crate::closure::storage::ClosureFn;
+use crate::closure::storage::OptionalFmMutClosure;
+
+
+
+// ============
+// === Slot ===
+// ============
+
+/// A single event listener slot.
+///
+/// Stores a closure that can be registered as an event listener.
+///
+/// Slot will register the closure as event listener whenever both closure and target are provided.
+///
+/// Caveat: this listener holds a reference to the target while it is registered.
+/// Be sure not to leak this value nor have it dependent on target destruction.
```

Comment:
Ok, after reading the coe I dont really understand the mechanism of `Event` and `Slot`. The docs should explain it better + show examples of its usage.